misk-crypto/src/main/kotlin/misk/crypto/pgp/PgpDecrypterProvider.kt

```diff
@@ -0,0 +1,59 @@
+package misk.crypto.pgp
+
+import com.google.crypto.tink.KmsClient
+import com.google.inject.Inject
+import com.google.inject.Provider
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import misk.crypto.Key
+import misk.crypto.KeyReader
+import misk.moshi.adapter
+import org.bouncycastle.openpgp.PGPSecretKeyRingCollection
+import org.bouncycastle.openpgp.PGPUtil
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder
+import java.util.Base64
+
+internal class PgpDecrypterProvider(
+  private val key: Key,
+  private val kmsUri: String?,
+  private val isReal: Boolean
+) : Provider<PgpDecrypter>, KeyReader() {
+  @Inject private lateinit var kmsClient: KmsClient
+
+  override fun get(): PgpDecrypter {
+    val jsonAdapter = moshi.adapter<PgpKeyJsonFile>()
+    val pgpKeyJsonFile =
+        jsonAdapter.fromJson(key.encrypted_key.value) ?: error("could not deserialize json file")
+
+    val decoded = Base64.getDecoder().decode(pgpKeyJsonFile.encrypted_private_key)
```

Comment:
i could do `pgpKeyJsonFile.encrypted_private_key.decodeBase64()!!.toByteArray()` i guess, though it doesn't really read much better

Reply:
I guess you still need it as a byte array in the end, so not really saving anything. Unclear to me whether this would provide any cleaner error handling.